integration/__tests__/upgrade_to_m10__hardfork_scen.js

```diff
@@ -0,0 +1,185 @@
+          "miner": "0xbb7b8287f3f0a933474a79eae42cbca977791171",
+          "mixHash": "0x4fffe9ae21f1c9e15207b1f472d5bbdd68c9595d461666602f2be20daf5e7843",
+          "nonce": "0x689056015818adbe",
+          "number": "0x1",
+          "parentHash": block0.hash,
+          "receiptsRoot": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
+          "sha3Uncles": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
+          "size": "0x220",
+          "stateRoot": "0xddc8b0234c2e0cad087c8b389aa7ef01f7d79b2570bccb77ce48648aa61c904d",
+          "timestamp": "0x55ba467c",
+          "totalDifficulty": "0x78ed983323d",
+          "transactions": [
+          ],
+          "transactionsRoot": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
+          "uncles": [
+          ]
+        }
+      };
+
+      let i = 1;
+      async function expectStuck(actor) {
+        console.log(`DFR expectStuck ${i++} ${actor.name}`);
+        let lastProcessedBlock0 = (await actor.api.query.cash.lastProcessedBlock('Eth')).toJSON();
+        let pendingChainBlocks0 = (await actor.api.query.cash.pendingChainBlocks('Eth')).toJSON();
+
+        expect(lastProcessedBlock0.eth).toEqual({
+          hash: block0.hash,
+          parent_hash: '0x0000000000000000000000000000000000000000000000000000000000000000',
+          number: 0,
+          events: []
+        });
+
+        expect(pendingChainBlocks0[0]).toEqual({
+          block: {
+            eth: {
+              hash: block1.hash,
+              parent_hash: block0.hash,
+              number: 1,
+              events: []
+            }
+          },
+          support: [
+            '0x8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48',
+            '0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d'
+          ],
+          dissent: [
+            '0x306721211d5404bd9da88e0204360a1a9ab8b87c66c1bc2fcdd37f3c2222cc20',
+            '0x90b5ab205c6974c9ea841be688864633dc9ca8a357843eeacf2314649965fe22'
+          ]
+        });
+      }
+
+      async function expectUnstuck(actor) {
+        console.log(`DFR expectUnstuck ${i++} ${actor.name}`);
+
+        let lastProcessedBlock1 = (await actor.api.query.cash.lastProcessedBlock('Eth')).toJSON();
+        let pendingChainBlocks1 = (await actor.api.query.cash.pendingChainBlocks('Eth')).toJSON();
+
+        expect(lastProcessedBlock1.eth.number).toBeGreaterThan(5); // Expect progress
+        expect(pendingChainBlocks1).toEqual([]); // Lots of progress
+      }
+
+      let proxyJson = (json) => {
+        return (req, res, body) => {
+          res.setHeader("Content-Type", "application/json");
+          res.writeHead(200);
+          res.end(JSON.stringify(json, null, 4));
+        };
+      };
+
+      [charlie.ethProxy, dave.ethProxy].forEach((proxy) => {
+        proxy.replace(/eth_getBlockByNumber.*\"0x1\"/, proxyJson(badBlock1));
```

Comment:
This seems like it'll be a bit fragile using a regex on the raw body instead of matching fields, if the order in the JSON changes it'll just silently stop replacing?